File: Data-Structures/Stack/stack.cpp
//Implementation of stack
#include<iostream>

using namespace std;
const int MAX=10;

class stack {

    private:
        int arr[MAX];
        int top;

    public:
        stack() {
            top = -1;
        }

        void push(int item) {
            if(top == MAX -1) {
            cout << endl << "Array is full" ;
            return;
            }

            top ++;
            arr[top] = item;
        }

        int pop() {
            if(top == -1) {
                cout << endl << "Array is empty";
                return 0;
            }

            int data =arr[top];
            top--;

            return data;
        }

};

int main() {

    stack s;
    int n = 3;
    cout << "Enter three elements to enter into ";
    cout << "the stack" << endl;
    while(n) {
        int i;
        cin >> i;
        s.push(i);
        n--;
        }

    int i = s.pop();
    cout << "Item removed is " << i;

    return 0;
}

File: Data-Structures/stack.py
class ArrayStack:

 def init (self):
#Create an empty stack
 self. data = [ ] # nonpublic list instance

 def len (self):
#Return the number of elements in the stack.”””
 return len(self. data)

 def is empty(self):
#Return True if the stack is empty.”””
return len(self. data) == 0

 def push(self, e):
#Add element e to the top of the stack.”””
 self. data.append(e) # new item stored at end of list
 def top(self):
#Return (but do not remove) the element at the top of the stack.
 #Raise Empty exception if the stack is empty.


 if self.is empty( ):
 raise Empty( Stack is empty )
 return self. data[−1] # the last item in the list
 def pop(self): #Remove and return the element from the top of the stack (i.e., LIFO).
 #Raise Empty exception if the stack is empty.
 
 if self.is empty( ):
 raise Empty( Stack is empty )
 return self. data.pop( ) # remove last item from list
